core/iwasm/aot/aot_runtime.c

```diff
@@ -1684,8 +1686,14 @@ aot_instantiate(AOTModule *module, AOTModuleInstance *parent,
     if (stack_size == 0)
         stack_size = DEFAULT_WASM_STACK_SIZE;
 #if WASM_ENABLE_SPEC_TEST != 0
+#if WASM_ENABLE_TAIL_CALL == 0
     if (stack_size < 128 * 1024)
         stack_size = 128 * 1024;
+#else
+    /* Some tail-call cases require large operand stack */
```

Comment:
which test cases?

Reply:
Some opcode return_call_indirect and return_call related cases which are added in gc spec proposal repo (and are not added in the default [spec](https://github.com/WebAssembly/spec/tree/main/test/core) repo):
https://github.com/WebAssembly/gc/blob/master/test/core/return_call_indirect.wast#L260-L261
https://github.com/WebAssembly/gc/blob/master/test/core/return_call_indirect.wast#L266-L267
https://github.com/WebAssembly/gc/blob/master/test/core/return_call.wast#L105
https://github.com/WebAssembly/gc/blob/master/test/core/return_call.wast#L111-L112
Since in the LLVM IR translated, we recursively call the function but not re-use the native stack (and operand stack if AOT/JIT stack frame is enabled) to call it, it requires lots of native/operand stack. Maybe we can optimize this in the future?